### PR TITLE
Merge from master

### PR DESCRIPTION
**Pull request by ikostan**
## Summary by Sourcery

This pull request includes a new kata solution, test improvements, refactorings for readability, and updates to the build and CI pipelines. It introduces a solution for the 'Replace With Alphabet Position' kata and enhances existing tests with parameterized tests and edge cases. The build process is updated to use Python 3.12 and Ubuntu 24.04, and includes new PyType linting workflows. The CI pipeline is configured to ignore specific branches for pull requests and to run pytest after PyType linting.

New Features:
- Adds a solution for the 'Replace With Alphabet Position' kata.
- Adds tests for the 'Replace With Alphabet Position' kata, including edge cases.

Enhancements:
- Refactors the Sudoku test cases to use parameterized tests for better readability and maintainability.
- Refactors the is_prime test cases to use parameterized tests for better readability and maintainability.
- Refactors the gen_primes test cases to use parameterized tests for better readability and maintainability.
- Improves the readability of the 'make_spiral' solution by extracting conditional logic into separate functions.
- Improves the readability of the 'battleship_field_validator' solution by using the all() function.
- Simplifies the calculation of combinations in the 'easy_line' solution.
- Adds type hints to the `__calc_rgb` method in the `Potion` class.

Build:
- Updates GitHub Actions workflows to use Python 3.12 and Ubuntu 24.04.
- Updates GitHub Actions workflows to install dependencies using `python -m pip`.
- Updates GitHub Actions workflows to run linters and type checkers using `python -m`.
- Updates GitHub Actions workflows to include a timeout of 10 minutes.
- Updates GitHub Actions workflows to run pytest with coverage reporting.
- Updates GitHub Actions workflows to report coverage to Codacy.
- Adds new GitHub Actions workflows for PyType linting for kyu2, kyu3, kyu6, kyu7, and kyu8.
- Removes the workflow for MyPy and replaces it with individual workflows for each kyu.
- Removes the workflow for PyType and replaces it with individual workflows for each kyu.
- Adds a workflow for testing the utils directory.

CI:
- Configures the main build pipeline to ignore the master branch and kyu branches for pull requests.
- Configures the main build pipeline to run PyType linting for kyu2, kyu3, kyu6, kyu7, and kyu8.
- Configures the main build pipeline to run pytest after PyType linting.

Tests:
- Adds test cases for empty strings, numbers, special characters, and mixed alphanumeric input to the 'Replace With Alphabet Position' kata.
- Adds a test case for the print_log function.

Chores:
- Removes the pytest skip mark from the line_safari_is_that_a_line test.
- Removes the copy_allure_history.py script.

### CODE REVIEW

File: kyu_3/make_spiral/solution.py [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
 Created by Egor Kostan.
 GitHub: https://github.com/ikostan
 """
 
-
 DIRECTIONS: dict = {
     'right': 'down',
     'down': 'left',
     'left': 'up',
-    'up': 'right'
```
[the remark this answers is not in the document]
Comment (ikostan) [reply kyu_3/make_spiral/solution.py:14]: Will do it when work on kyu_3 branch
